### ISSUE TITLE
MSD docstring inconsistent with class inputs (select)

### ISSUE THREAD

**Issue by IAlibay** [MDAnalysis/mdanalysis#2784]
## Expected behavior ##

According to the docstring `select` defaults to `None` so not passing anything should just work.
https://github.com/MDAnalysis/mdanalysis/blob/333bb04e3a730278aa49ff4c6db372e800f05cd6/package/MDAnalysis/analysis/msd.py#L279-L281

## Actual behavior ##

`select` is a required positional argument.

## Code to reproduce the behavior ##

``` python
import MDAnalysis as mda
from MDAnalysis.analysis.msd import EinsteinMSD as MSD
from MDAnalysis.tests.datafiles import PSF, DCD

u = mda.Universe(PSF, DCD)

m = MSD(u)
```

## Current version of MDAnalysis ##

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`) 2.0.0-dev0
- Which version of Python (`python -V`)? 3.7.7
- Which operating system? Ubuntu 20.04 LTS (WSL2)

**Comment by lilyminium** [comment]
IMO this class is most flexible when `u` can be either a Universe or an AtomGroup and `select='all'` is the default.

**Comment by orbeckst** [comment]
@lilyminium ‘s suggestion sounds sensible. It shifts the usage pattern towards “work with AtomGroups”. (If the emphasis were on starting from universe then having a required `select` makes more sense to me as ‘all’ is probably not a good default.)

Will the class work with updating AGs, though? And if not, then it should raise an error (TypeError I think).

**Comment by IAlibay** [comment]
> Will the class work with updating AGs, though? And if not, then it should raise an error (TypeError I think).

Not in its current form, the MSD code relies on n_atoms being known in the class construction stage: https://github.com/MDAnalysis/mdanalysis/blob/333bb04e3a730278aa49ff4c6db372e800f05cd6/package/MDAnalysis/analysis/msd.py#L303

Since updating atom groups can have variable lengths, the current code might fail somewhere around here:
https://github.com/MDAnalysis/mdanalysis/blob/333bb04e3a730278aa49ff4c6db372e800f05cd6/package/MDAnalysis/analysis/msd.py#L342-L343

Either way, I could be wrong but I don't think updating atomgroups make sense in terms of MSDs.

pinging @hmacdope here.

**Comment by hmacdope** [comment]
Hey all, thanks for spotting this and sorry for so many bugs :/.

I agree with @IAlibay in that updating atom groups don't make much sense for MSDs. As it stands the current code cannot handle it so we should enforce no updating AGs.

**Comment by IAlibay** [comment]
> Hey all, thanks for spotting this and sorry for so many bugs :/.

No need to apologise, the nature of code development is that bugs happen :)
I'd be more concerned if code never happened to have any bugs 🤣 

> I agree with @IAlibay in that updating atom groups don't make much sense for MSDs. As it stands the current code cannot handle it so we should enforce no updating AGs.

The fix here is quite trivial, do you to fix it @hmacdope? Otherwise I'm sure someone can pick it up later (probably worth waiting until #2783 gets merged first though).

**Comment by hmacdope** [comment]
Ill do it post #2783.

**Comment by orbeckst** [comment]
It is post #2783 😁
